bundles/binding/org.openhab.binding.cardio2e/src/main/java/org/openhab/binding/cardio2e/internal/code/Cardio2eDecoder.java

```diff
@@ -0,0 +1,831 @@
+						decodeComplete = true;
+					}
+				} catch (IllegalArgumentException e1) {
+					e1.printStackTrace();
+				} catch (Exception e) {
+					e.printStackTrace();
+				}
+			}
+			break;
+		}
+		return decodeComplete;
+	}
+
+	private boolean decodeHvacControlParameters(Cardio2eHvacControlTransaction receivedCardio2eTransaction, List<String> digestedTransaction) {
+		boolean decodeComplete = false;  
+		switch (receivedCardio2eTransaction.getTransactionType()){
+		case ACK:
+		case GET:
+			if (digestedTransaction.size() == 3) {
+				try {
+					receivedCardio2eTransaction.setObjectNumber (Short.parseShort(digestedTransaction.get(2)));
+					//System.out.print("#"+receivedCardio2eTransaction.getObjectNumber()); //Testing output
+					decodeComplete = true;
+				} catch (IllegalArgumentException e1) {
+					e1.printStackTrace();
+				}
+			}
+			break;
+		case NACK:
+			decodeComplete = decodeNACK(receivedCardio2eTransaction, digestedTransaction);
+			break;
+		case INFORMATION:
+		case SET:
+			if (digestedTransaction.size() == 7) {
+				try {
+					if ((digestedTransaction.get(5).length()==1) && (digestedTransaction.get(6).length()==1)) {
+						receivedCardio2eTransaction.setObjectNumber (Short.parseShort(digestedTransaction.get(2)));
+						receivedCardio2eTransaction.setHvacHeatingSetPoint(Double.parseDouble(digestedTransaction.get(3)));
+						receivedCardio2eTransaction.setHvacCoolingSetPoint(Double.parseDouble(digestedTransaction.get(4)));
+						receivedCardio2eTransaction.setHvacFanState (Cardio2eHvacFanStates.fromSymbol(digestedTransaction.get(5).charAt(0)));
+						receivedCardio2eTransaction.setHvacSystemMode (Cardio2eHvacSystemModes.fromSymbol(digestedTransaction.get(6).charAt(0)));
+						//System.out.print("#"+receivedCardio2eTransaction.getObjectNumber()+" HEATING / COOLING SETPOINTS: "+receivedCardio2eTransaction.getHvacHeatingSetPoint()+" / "+receivedCardio2eTransaction.getHvacCoolingSetPoint()+"ºC FAN "+receivedCardio2eTransaction.getHvacFanState()+" MODE "+receivedCardio2eTransaction.getHvacSystemMode()); //Testing output
+						decodeComplete = true;
+					}
+				} catch (IllegalArgumentException e1) {
+					e1.printStackTrace();
+				} catch (Exception e) {
+					e.printStackTrace();
+				}
+			}
+			break;
+		}
+		return decodeComplete;
+	}
+
+	@SuppressWarnings("incomplete-switch")
+	private boolean decodeScenarioParameters(Cardio2eScenarioTransaction receivedCardio2eTransaction, List<String> digestedTransaction) {
+		boolean decodeComplete = false;  
+		switch (receivedCardio2eTransaction.getTransactionType()){
+		case ACK:
+			if (digestedTransaction.size() == 3) {
+				try {
+					receivedCardio2eTransaction.setObjectNumber (Short.parseShort(digestedTransaction.get(2)));
+					//System.out.print("#"+receivedCardio2eTransaction.getObjectNumber()); //Testing output
+					decodeComplete = true;
+				} catch (IllegalArgumentException e1) {
+					e1.printStackTrace();
```

Comment:
Use logger.error() instead of e.printStackTrace()

Reply:
Uploaded changes.